tests/ops/op_math/test_pow_op.py

```diff
@@ -744,6 +744,15 @@ def test_matrix_against_shortcut_tf(self, z):
         param = tf.Variable(2.34)
         assert self.check_matrix(param, z)
 
+    @pytest.mark.tf
```

Comment:
What I'm mildly confused about is why `test_matrix_against_shortcut_tf` was passing.  `2` is an integer...

Reply:
it's because that tester function uses `qml.matrix()`, and that [uses the decomposition](https://github.com/PennyLaneAI/pennylane/blob/768bea0c044cb38efe342bca7fa3a124ecfb03bd/pennylane/ops/functions/matrix.py#L140-L143) if `op.matrix()` fails.
```pycon
>>> qml.pow(qml.IsingZZ(tf.Variable(2.34), wires=(0, 1)), 2).decomposition()
[IsingZZ(<tf.Tensor: shape=(), dtype=float32, numpy=4.68>, wires=[0, 1])]
```

Reply:
Ah... mind changing that to `op.matrix` instead of `qml.matrix(op)` so we catch that in the future?